### PR TITLE
Add scenarios in integration tests relevant to lisk-p2p

### PR DESCRIPTION
**Pull request by ishantiw**
### Description

Add missing integration tests on lisk-p2p. Cover different scenarios like, for example, when a couple of nodes become unresponsive due to some reason, other nodes connected should be able to detect and update their network status gracefully. 

### Review checklist

* The PR resolves #INSERT_ISSUE_NUMBER
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: packages/lisk-p2p/test/integration/p2p.ts [hunk 1]
```diff
@@ -46,9 +47,9 @@ describe('Integration tests for P2P library', () => {
 		afterEach(async () => {
 			await Promise.all(
 				p2pNodeList.map(async p2p => {
-					try {
+					if (p2p.isActive) {
```
Comment (shuse2) [inline packages/lisk-p2p/test/integration/p2p.ts]: maybe filter first?
`p2pNodeList.filter(p2p => p2p.isActive).map(xxx`

File: packages/lisk-p2p/test/integration/p2p.ts [hunk 2]
```diff
@@ -198,15 +211,177 @@ describe('Integration tests for P2P library', () => {
 				await wait(DISCOVERY_INTERVAL * 5);
 
 				p2pNodeList.forEach(p2p => {
-					let {connectedPeers} = p2p.getNetworkStatus();
+					let { connectedPeers } = p2p.getNetworkStatus();
```
Comment (shuse2) [inline packages/lisk-p2p/test/integration/p2p.ts]: `const` here?

File: packages/lisk-p2p/test/integration/p2p.ts [hunk 3]
```diff
@@ -58,6 +59,18 @@ describe('Integration tests for P2P library', () => {
 				expect(p2p).to.have.property('isActive', true);
 			});
 		});
+
+		describe('P2P.request', () => {
+			it('should throw an error when not able to get any peer in peer selection', async () => {
```
Comment (shuse2) [inline packages/lisk-p2p/test/integration/p2p.ts:62]: does user have control over if peer can be obtained in selection or not?
or was if it at least has seed, this one happen?

**Review by shuse2 — CHANGES_REQUESTED** [review]

File: packages/lisk-p2p/test/integration/p2p.ts (hunk 3 above)
Comment (ishantiw) [reply packages/lisk-p2p/test/integration/p2p.ts:62]: No, user doesn't have control over the default peer selection but I think that needs to be covered as well where the user is passing its own peer selection function. But here, we are just testing in a completely disconnected network with no peers and seed nodes, we expect it to throw this error.

**Comment by ishantiw** [comment]
I will add test cases related to the custom selection function. Issue created #1131

**Review by shuse2 — APPROVED** [review]

**Review by jondubois — APPROVED** [review]

**Pull request merged**